pedsnetdcc/indexes.py

```diff
@@ -0,0 +1,90 @@
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.schema import CreateIndex, DropIndex
+
+from pedsnetdcc import VOCAB_TABLES, TRANSFORMS
+from pedsnetdcc.utils import stock_metadata
+
+
+def _indexes_from_metadata(metadata, transforms, vocabulary=False):
+    """Return list of SQLAlchemy index objects for the transformed metadata.
+
+    Given the stock metadata, for each transform `T` we invoke:
+
+        new_metadata = T.modify_metadata(metadata)
+
+    and at the end, we extract the indexes.
+
+    :param metadata: SQLAlchemy metadata for PEDSnet
+    :type: sqlalchemy.schema.MetaData
+    :param transforms: list of Transform classes
+    :type: list(type)
+    :param vocabulary: whether to return indexes for vocabulary tables or
+    non-vocabulary tables
+    :type: bool
+    :return: list of index objects
+    :rtype: list(sqlalchemy.Index)
+    """
+    for t in transforms:
+        metadata = t.modify_metadata(metadata)
+
+    indexes = []
+    for name, table in metadata.tables.items():
+        if vocabulary:
+            if name in VOCAB_TABLES:
+                indexes.extend(table.indexes)
+        else:
+            if name not in VOCAB_TABLES:
+                indexes.extend(table.indexes)
+
+    return indexes
+
+
+def _indexes_from_model_version(model_version, vocabulary=False):
```

Comment:
Yes, you're right. I wrote it first and then decided I wanted to refactor to provide a function accepting the metadata as a parameter. After the refactor, the original function no longer adds any value, especially since the `stock_metadata` makes it so easy to create the necessary metadata. I am removing.